playground/frontend/lib/modules/examples/examples_dropdown_content.dart

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:playground/constants/links.dart';
+import 'package:playground/constants/sizes.dart';
+import 'package:playground/modules/examples/components/example_list/example_list.dart';
+import 'package:playground/modules/examples/components/filter/filter.dart';
+import 'package:playground/modules/examples/components/search_field/search_field.dart';
+import 'package:playground_components/playground_components.dart';
+import 'package:url_launcher/url_launcher.dart';
+
+class ExamplesDropdownContent extends StatefulWidget {
+  final VoidCallback onSelected;
+  final PlaygroundController playgroundController;
+
+  const ExamplesDropdownContent({
+    required this.onSelected,
+    required this.playgroundController,
+  });
+
+  @override
+  State<ExamplesDropdownContent> createState() =>
+      _ExamplesDropdownContentState();
+}
+
+class _ExamplesDropdownContentState extends State<ExamplesDropdownContent> {
+  final _searchController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    switch (widget.playgroundController.exampleCache.catalogStatus) {
+      case LoadingStatus.error:
+        return const LoadingErrorWidget();
+
+      case LoadingStatus.loading:
+        return const LoadingIndicator();
+
+      case LoadingStatus.done:
+        return _buildLoaded();
+    }
+  }
+
+  Widget _buildLoaded() {
```

Comment:
Yes, mostly for performance reasons. The only official explanation is here: https://www.youtube.com/watch?v=IOyq-eTRhvo

Reply:
Good point for many cases. But here we should not split this. Whenever this is rebuilt we should also check the catalog status to maybe go into another `case`. Splitting this will require an extra `AnimatedBuilder` on `playgroundController`. This helper method actually puts its disadvantage into use.